app/src/Controller/LdapController.php

```diff
@@ -0,0 +1,247 @@
+<?php
+
+namespace App\Controller;
+
+use App\Service\Ldap\Client;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Ldap\Entry;
+use Symfony\Component\Ldap\Exception\LdapException;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\Serializer\SerializerInterface;
+use App\DTO\LdapEntryDTO;
+use Symfony\Component\ErrorHandler\ErrorRenderer\SerializerErrorRenderer;
+use Symfony\Component\Serializer\Exception\NotEncodableValueException;
+
+class LdapController extends AbstractController
+{
+    /**
+     * @Route("/api/ldap", name="get_ldap_entries", methods={"GET"})
+     *
+     * @return JsonResponse
+     */
+    public function getLdapEntries(
+        Client $ldap,
+        Request $request,
+        SerializerInterface $serializer
+    ): JsonResponse {
+
+        $query = $request->get('query', '(objectClass=*)');
+        $baseDn = $request->get('base', null);
+
+        $options = [];
+
+        $filters = json_decode($request->get('filters', '[]'), true);
+        if (!empty($filters)) {
+            $queryFilters = '';
+            foreach ($filters as $field => $value) {
+                switch ($field) {
+                    case 'dn':
+                        $queryFilters .= "($value)";
+                        break;
+
+                    default:
+                        $queryFilters .= "($field=$value)";
+                        break;
+                }
+            }
+            $query = "(&$queryFilters$query)";
+        }
+        // XXX Is there any way to ask LDAP server to sort results?
+        //$orders  = json_decode($request->get('orders', null), true);
+
+        // Attributes option.
+        $attributes = $request->get('attributes', ['dn']);
+        $options['filter'] = $attributes;
+
+        // Add query options.
+        $size = (int) $request->get('size', 0);
+        $max = (int) $request->get('max', 0);
+
+        $options['pageSize'] = $size;
+        $options['maxItems'] = $max;
+
+        $ldap->bind();
+
+        //XXX Verify If there is a need to throw a special error by try catch
+        $ldapEntries = $ldap->search($query, $baseDn, $options);
+        $total = count($ldapEntries);
+        $entries = array();
+
+        foreach ($ldapEntries as $key => $ldapEntry) {
+            $entries[$key]['dn'] = $ldapEntry->getDn();
+
+            $ldapDto = new LdapEntryDTO();
+            //XXX Verify if there is a need to throw a special error by try catch
+            $ldapEntry =  $ldapDto->serializeJpegPhoto($ldapEntry);
```

Comment:
Clearly needs to be improved. Creating a new `LdapEntryDTO` to call a static method makes no sense!